ext/filter/filter.stub.php

```diff
@@ -4,17 +4,13 @@
 
 function filter_has_var(int $type, string $variable_name): bool {}
 
-/** @param array|int $options */
-function filter_input(int $type, string $variable_name, int $filter = FILTER_DEFAULT, $options = null): mixed {}
+function filter_input(int $type, string $variable_name, int $filter = FILTER_DEFAULT, array|int|null $options = null): mixed {}
```

Comment:
Is it not possible to make the default `0` or `[]` or so and forego the null value?

Reply:
oh yes, they are pretty much possible to be changed!

Reply:
Actually, it's not possible - or at least I'm unsure about it. E.g. there is a code in `php_filter_call()`:

```if (filter_args && Z_TYPE_P(filter_args) != IS_ARRAY) {
    zend_long lval = filter_args_long;
    if (filter != -1) { /* handler for array apply */
    /* ... */
```

And especially this piece of code seems problematic, even if we would go with the 0 default value.